File: backend/src/main/java/com/faruk/backend/entity/Product.java
package com.faruk.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor // generise konstruktor bez arg
@AllArgsConstructor // generise sa svim arg
@Table(name="products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String sku;

    @Column(nullable = false)
    private String name;

    private String description;

    private String imageUrl;

    @Column(nullable = false)
    private BigDecimal price;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="category_id", nullable = false)
    private Category category;

    private Long unitsInStock;

    private Date datePublished;

    private Date dateUpdated;
}

File: frontend/src/app/models/product.ts
import { Category } from "./category";
import { User } from "./user";

export interface Product{
    id: number;
    sku: string;
    name: string;
    description: string;
    imageUrl: string;
    price: number;
    category: Category;
    user: User;
    unitsInStock: number;
    datePublished?: Date;
    dateUpdated?: Date;
}